shell/models/secret.js

```diff
@@ -365,33 +366,49 @@ export default class Secret extends SteveModel {
     return (key, value) => { // or (mapOfNewData)
       const isMap = key && typeof key === 'object';
 
-      if ( !this.data || isMap ) {
+      if (!this.data || isMap) {
         set(this, 'data', {});
       }
 
       let neu;
 
-      if ( isMap ) {
+      if (isMap) {
         neu = key;
       } else {
         neu = { [key]: value };
       }
 
-      for ( const k in neu ) {
+      for (const k in neu) {
         // The key is quoted so that keys like '.dockerconfigjson' that contain dot don't get parsed into an object path
         set(this.data, `"${ k }"`, base64Encode(neu[k]));
       }
     };
   }
 
   get doneRoute() {
-    if ( this.$rootGetters['currentProduct'].name === MANAGER ) {
+    if (this.$rootGetters['currentProduct'].name === MANAGER) {
       return 'c-cluster-manager-secret';
     } else {
       return 'c-cluster-product-resource';
     }
   }
 
+  cleanForSave(data) {
+    // There are some properties in ACTIVELY_REMOVE
+    // that should be removed from the data
+    const val = super._cleanForSave(data);
+
+    const fieldsToRemove = [...NEVER_ADD];
```

Comment:
I did not write the code, still I kept this because possible modifications to an object affect other imports.
An Enum would be probably preferred but that's not in our architecture.
This highlights some issues with our configuration which we should tackle down.

Reply:
removed non-required spread